packages/pages/src/dev/server/ssr/yextEnv.ts

```diff
@@ -28,34 +28,23 @@ export function getEnvFromYextCredential(): YextEnv {
       return "production";
     }
 
-    if (
-      typeof data === "object" &&
-      data !== null &&
-      "env" in data &&
-      typeof (data as any).env === "string"
-    ) {
-      const env = (data as { env: string }).env as string;
-
-      const validEnvs: YextEnv[] = [
-        "development",
-        "qa",
-        "sandbox",
-        "production",
-      ];
-      if (validEnvs.includes(env as YextEnv)) {
+    const env = (data as { env: string }).env as string;
+    if (env) {
```

Comment:
There are still way too many conditions here for a simple _return value if exists and is valid, otherwise return a default_.